proxy/src/lib.rs

```diff
@@ -172,9 +199,70 @@ where
             outbound_listener,
             metrics_listener,
             get_original_dst,
-            reactor: mut core,
         } = self;
 
+
+
+        // Our "pool" of one worker thread will use this reactor.
```

Comment:
I discussed why we have to use a single-thread "pool" in https://github.com/runconduit/conduit/pull/916/files#r186901911

Reply:
1. I think if we don't store `Sender`s and whatnot, and use `tokio::spawn` instead, then we can use either runtime mode.
2. Can this just use the `tokio::runtime` builder instead?